lib/src/utils.dart

```diff
@@ -499,6 +503,113 @@ String yamlToString(data) {
   return buffer.toString();
 }
 
+/// [YamlMap] which overrides entries in the [original] map with entries from
+/// the [overrides] map.
+///
+/// This map is useful to apply overrides from one yaml file to another, while
+/// keeping source references intact.
+///
+/// The map contains entries for all of the keys of [original]. If both
+/// [original] and [overrides] contain an entry for a given key, [mergeEntry]
+/// is called to merge the values. Otherwise, the entry from [original] is taken
+/// over as is.
+///
+/// Entries that only exist in [overrides] are taken over as is.
+///
+/// The [YamlNode]s of keys of merged entries always come from [original].
+class OverrideYamlMap with MapMixin, UnmodifiableMapMixin implements YamlMap {
```

Comment:
I think it is better to just build the merged map explicitly...

Reply:
Please see https://github.com/dart-lang/pub/pull/3215#discussion_r779486656